Source/Provers/LeanAuto/LeanAutoGenerator.cs

```diff
@@ -364,17 +364,21 @@ public override Expr VisitLambdaExpr(LambdaExpr node)
 
   public override Expr VisitLetExpr(LetExpr node)
   {
-    if (node.Dummies.Count > 1) {
+    if (node.Dummies.Count != node.Rhss.Count) {
       throw new LeanConversionException(node.tok,
-        "Unsupported: LetExpr with more than one binder");
+        "Unsupported: LetExpr with differing LHS and RHS counts.");
+    }
+
+    var bindings = node.Dummies.Zip(node.Rhss);
+    foreach (var (x, e) in bindings) {
+      WriteText("(let ");
+      Visit(x.TypedIdent);
+      WriteText(" := ");
+      Visit(e);
+      WriteText("; ");
     }
-    WriteText("(let");
-    node.Dummies.ForEach(x => Visit(x.TypedIdent));
-    WriteText(" := ");
-    node.Rhss.ForEach(e => Visit(e));
-    WriteText("; ");
     Visit(node.Body);
-    WriteText(")");
+    bindings.ForEach(b => WriteText(")"));
```

Comment:
IMO `.ForEach` is an anti-pattern, since the IDE and debugger support is worse than when using the language-construct `foreach`